crowbar_framework/app/models/service_object.rb

```diff
@@ -978,6 +978,7 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     else
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
+      new_elements = skip_unready_nodes(@bc_name, inst, new_elements)
```

Comment:
You need to pass `old_role` (or `old_elements` actually, but that means moving a block of code) here: if a node was not in the previous deployment, then we should not skip it when it's unready as it means the apply was partly about adding that node.

Reply:
Thing is, if we skip below this lines then we still get the same issue, and that is that the proposal gets queued as there are nodes not ready, thus blocking the proposal deployment until those unready nodes are back online, which is the point of the card attached to this.

Now that I think about it, maybe a better solution would be just to skip showing the unready nodes in the ui list of the barclamp, but that would only work trougth UI, not trougth CLI.

Maybe a different approach is needed for this ¬_¬

Reply:
Actually I can move the old_elements block here instead :)

Reply:
And actually looking at the code, and the logs we should not care if its `old_elements` or `new_elements` here.

If any `old_elements` are not in `new_elements`, they wont get checked, so no skip. And only elements in `new_elements` will be checked for ready-ness to queue the proposal so those are the ones that we want to skip.

Reply:
We should probably chat, because "then we still get the same issue, and that is that the proposal gets queued as there are nodes not ready," is actually what I think we should do if there's a node that was not in the proposal before.

Reply:
One other thing: we need to be super extra careful about one line that happens later on:

```
new_deployment["elements_expanded"] = new_elements
```

Right now, this would break because you would actually have only a subset of elements in there while we really really really want all elements, even the unready ones for that line. So you very likely need to use a second variable.